axon-cosmosdb-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/cosmosdb/springboot/autoconfig/CosmosAutoConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+package org.axonframework.extensions.cosmosdb.springboot.autoconfig;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import org.axonframework.common.ReflectionUtils;
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken;
+import org.axonframework.eventhandling.TrackingToken;
+import org.axonframework.eventhandling.tokenstore.TokenStore;
+import org.axonframework.extensions.cosmosdb.eventsourcing.tokenstore.CosmosTokenStore;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.json.JacksonSerializer;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.*;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jmx.support.RegistrationPolicy;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.containers.CosmosDBEmulatorContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.time.temporal.TemporalAmount;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Testcontainers
+class CosmosAutoConfigurationIntegrationTest {
```

Comment:
Although I understand that testing the `CosmosTokenStore` through a unit test didn't work ideal, we can still make a unit test for our auto-configuration. Especially with an eye on the fact the `CosmosDBEmulatorContainer` doesn't work on GitHub, I think this would be beneficial.

Reply:
Do we even need an integration test if the test is to check whether a bean is constructed with the right components?

Reply:
How can we unit test it without a client?

Reply:
Can you perhaps make a NoOp version of the client?

Reply:
No, since it's both a final class, and doesn't use an interface (besides `Closable`).